src/tdom/utils.py

```diff
@@ -12,12 +17,27 @@ def _as_comment(node):
 def _as_component(node, components):
   def component(value):
     def reveal():
-      props = { 'children': node['children'] }
-      for k, v in node['props'].items():
-        props[k] = v
+      if _IS_MICRO_PYTHON or ('children' in inspect.signature(value).parameters):
+        props = { 'children': node['children'] }
+        for k, v in node['props'].items():
+          props[k] = v
+      else:
+        props = node['props'].copy()
+
+      if _IS_MICRO_PYTHON:
+        try:  
+          result = value(**props)
+        except TypeError as e:
+          if "unexpected keyword argument" in str(e):
+            del props['children']
+            result = value(**props)
+          else:
+            raise e
+      else:
+        result = value(**props)
+
+      _replaceWith(node, _as_node(result))

```

Comment:
It's possible the flow for this could be improved, but that's minor. Looks good.